### DISCUSSION TITLE
First Layer Print Issues after Tramming/Mesh

### DISCUSSION THREAD

**Discussion by scurker** [mriscoc/Ender3V2S1#746] [Q&A]
So I recently added a CRTouch to my Ender 3 V2, and so far I've been unsuccessful at getting the bed to print the first layer correctly. I've run the tramming wizard to get the tolerances with ~0.01, and then ran the auto build mesh after and saved it to slot 0. Then adjusted the Z-Offset accordingly. But no matter what I try, I can't seem to get a level print. The left side of the bed is _always_ slightly too close to the extruder tip, and the right side is _always_ slightly too far leading to there being poor contact with the 1st layer on the right side.

I've repeated the above steps multiple times after checking all the various screws and making sure nothing wobbles. Am I doing something wrong? I had better success prior to having the CRTouch installed, so I don't know if I'm doing something wrong or if something else is messed up.

Here's my start GCode:

```
G92 E0 ; Reset Extruder
G28 ; Home all axes
G29 A ; activate leveling
G29 L0 ; Load gmesh from slot 0
G29 J2 ; tilt mesh
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X0.1 Y20 Z0.3 F5000.0 ; Move to start position
G1 X0.1 Y200.0 Z0.3 F1500.0 E15 ; Draw the first line
G1 X0.4 Y200.0 Z0.3 F5000.0 ; Move to side a little
G1 X0.4 Y20 Z0.3 F1500.0 E30 ; Draw the second line
G92 E0 ; Reset Extruder
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X5 Y20 Z0.3 F5000.0 ; Move over to prevent blob squish
```

I'm on the latest firmware, `Ender3V2-422-BLTUBL-MPC-20221222.bin`

**Comment by Jimidean23** [comment]
I was having similar issues after my CR touch install. 

I opted to not do the tilt probe before each.  I don't know why. But that always seem to throw off prints.  

As for the probe.  It was just a hair too far from the bed.  I found that it was hitting the nozzle first and throwing off the mesh.  Anyway. I read another guy had same issue and he added 1mm washers to the cr touch to bring it closer to bed just a little. And that worked perfect for me too.  

From there.  I did the 4 corner manual tramming with a 2mm filler gauge.  Just to get it level.  Didn't use the wizard. 

Then used the z wizard and a paper receipt for gap not a regular paper. 

Created a mesh. And done deal.  I have been printing for a month since with about 300hrs of printing and haven't had to touch anything.  

Again. This worked for me.  Might be diff for others.  Just sharing my fix.

**Comment by scurker** [comment]
> I opted to not do the tilt probe before each. I don't know why. But that always seem to throw off prints.

Wow, this seems to be part of the problem. As soon as I turned this off I get a perfect 1st layer. I don't mind turning it off, but I was hoping to have a better understanding of why this doesn't seem to be working because having the tilt mesh turned on should be keeping the print consistent over time.

**Reply by ActiveAdminDirectory** [reply]
same issue as above whats your start gcode now then ?

**Reply by scurker** [reply]
I removed the following line:

`G29 J2 ; tilt mesh`

**Comment by Jimidean23** [comment]
I also opt to not do the tilt.  It throws off prints.  With it off.  It’s perfect every time.  I just have the activate and the L0 to load mesh.  Glad I’m not the only one who experienced this.  Sent from my iPhoneJimi Anthony BandaOn Mar 1, 2023, at 3:52 AM, AnthonyUDT ***@***.***> wrote:﻿
same issue as above whats your start gcode now then ?

**Comment by geeffland** [comment]
I am having the same issue... I'll try turning off G29 J2 also...

If that is turned off then I guess you just do a new mesh if the bed changes (i.e. painters tape, etc.)

Thanks,
Greg

**Reply by geeffland** [reply]
That seems to have helped.  I also went in and manually edited the mesh to lower some of the extremes .01 at a time for a few times and it seems to be working good now.

**Comment by ghost** [comment]
Shouldnt you activate the UBL after tilt/probe/load/autohome? So `G29 A` should be last one

**Reply by scurker** [reply]
Don't you need to activate leveling first before you can load from the mesh? That was my understanding as to why `G29 A` needs to come first.

**Comment by mriscoc** [comment]
The command `G29 A` enables the leveling system, only if a valid mesh is already loaded into the memory.
